### PR TITLE
Show countdown timer for last 7 days on the DP product page during a flash sale. 

### PR DESCRIPTION
**Pull request by lmath**
## Why are you doing this?

doesn't look terribly urgent

[**Trello Card**](https://trello.com/c/NyBumRDL/2134-countdown-timer-should-only-appear-during-the-last-week-of-the-print-sale)

## Changes

* Adds a function that determines if the now is x days from the end of the sale so that we can use it to determine if we should show the countdown timer
* Use this new function to only show the countdown timer 7 days before the end of the digital pack flash sale

## Screenshots
This is just changing the conditions by which we show the countdown timer or not, so n/a I think

### CODE REVIEW

**Comment by SarahLaughton** [comment]
👍 Love it!

File: assets/helpers/flashSale.js [hunk 1]
```diff
@@ -419,6 +419,16 @@ function getFormattedFlashSalePrice(
   return fixDecimals(sale.saleDetails[countryGroupId].price);
 }
 
+function showCountdownTimer(flashSaleActive: boolean, showForHowManyDays: number, endTime: number): boolean {
```
[the remark this answers is not in the document]
Comment (lmath) [reply assets/helpers/flashSale.js]: Good point. I will rename it.

File: assets/helpers/flashSale.js [hunk 2]
```diff
@@ -419,6 +419,16 @@ function getFormattedFlashSalePrice(
   return fixDecimals(sale.saleDetails[countryGroupId].price);
 }
 
+function countdownTimerActive(flashSaleActive: boolean, showForHowManyDays: number, endTime: number): boolean {
```
Comment (walaura) [inline assets/helpers/flashSale.js]: can we call this `countdownTimerIsActive`  (super pedantic i know)

File: assets/helpers/flashSale.js [hunk 3]
```diff
@@ -419,6 +419,16 @@ function getFormattedFlashSalePrice(
...
+  if (flashSaleActive) {
+    const now = new Date();
```
Comment (walaura) [inline assets/helpers/flashSale.js]:
```suggestion
    const timeTravelDays = getTimeTravelDaysOverride();
    const now = timeTravelDays ? Date.now() + (timeTravelDays * 86400000) : Date.now();
```

this'll allow the time travel parameter to work

**Pull request merged**